### PR TITLE
Eye Progress 1

### PR DESCRIPTION
**Pull request by ophthal**
Fixes #
Fix bold items - global search/replace turned them all into paragraphs, which they are not.
Fix Phamacy Search function in options.inc.php
Fix top panel layout.
Fix Font Awesome icons.
Fix loading spinner
Fix display patient Photograph.
Fix ACT box size
Add Copy/delete clinical values
Add in VITREOUS default values for DB.
Fix patient_caret
Got panelslider working again.
Fix navbar menu
@mytilene Retina QP fix in edit_list.php

TODO:
Fix pop-up Dispensing stuff/SpectacleRx
Fix jscolor (new version has deprecated lots of functions)
Fix Show IOP by Time
Fix Taskman (documents changes?)
Fix reports

This is functioning for the most part.
I will update this soon.  
Figured it is probably big enough to start review process - I can complete the TODO list concurrently.

### CODE REVIEW

File: interface/forms/eye_mag/php/taskman_functions.php [hunk 1]
```diff
@@ -36,52 +38,59 @@ function make_task($ajax_req)
 
     $query      = "SELECT * FROM users WHERE id=?";
     $to_data    =  sqlQuery($query, array($to_id));
-    $filename   = "Fax_" . $encounter . "_" . $to_data['lname'] . ".pdf";
+    $filename   = "Fax_".$enc."_".$to_data['lname'].".pdf";
 
-    $query = "SELECT * FROM documents where encounter_id=? and foreign_id=? and url like ? and deleted = 0";
-    $doc = sqlQuery($query, array($encounter,$pid,'%' . $filename . '%' ));
+    $query = "SELECT * FROM documents where encounter_id=? and foreign_id=? and url like ?";
```
Comment (bradymiller) [inline interface/forms/eye_mag/php/taskman_functions.php:43]: should add back the `and deleted = 0` part to this query or else will collect deleted documents

File: interface/forms/eye_mag/php/taskman_functions.php [hunk 2]
```diff
@@ -36,52 +38,59 @@ function make_task($ajax_req)
...
         // Delete the task, recreate the task, and send the newly made PDF.
         $sql = "DELETE from form_taskman where FROM_ID=? and TO_ID=? and PATIENT_ID=? and ENC_ID=?";
         $task = sqlQuery($sql, array($from_id,$to_id,$patient_id,$enc));
     }
-
     if ($task['ID'] && $task['COMPLETED'] == '2') {
-        $send['comments'] = xlt('This fax has already been sent.') . " " .
-                            xlt('If you made changes and want to re-send it, delete the original (in Communications) or wait 60 seconds, and try again.') . " " .
-                            xlt('Filename') . ": " . text($filename);
+        $send['comments'] = xlt('This fax has already been sent to')." ". $task['to_name'] ." via " .$task['to_fax']." on " .$sent_date.". ".
```
Comment (bradymiller) [inline interface/forms/eye_mag/php/taskman_functions.php:63]:
```
$send['comments'] = xlt('This fax has already been sent to') . " " . text($task['to_name']) . " via " . text($task['to_fax']) . " on " . text($sent_date) . ". " .
```

File: interface/forms/eye_mag/php/taskman_functions.php [hunk 3]
```diff
@@ -36,52 +38,59 @@ function make_task($ajax_req)
...
-								<i class='fa fa-file-pdf-o fa-fw'></i></a>
-								<i class='fa fa-repeat fa-fw'
-									onclick=\"top.restoreSession(); create_task('" . attr($pat_data['ref_providerID']) . "','Fax-resend','ref'); return false;\">
-									</i>
-							";
-                            //add a resend option.
+            $send['DOC_link'] = "<a href="JavaScript:void(0);"
+                                    onclick="openNewForm('" . $GLOBALS['webroot'] . "/controller.php?document&view&patient_id=" . attr($task['PATIENT_ID']) . "&doc_id=" . attr($task['DOC_ID']) . "', 'Fax Report');"
+                                    title='".xla('View the Summary Report sent to') . 
+                                            $task['to_name'] . " " . xla('via') . " " . $task['to_fax'] . " " . xla('on') . " " . $sent_date ."'>
```
Comment (bradymiller) [inline interface/forms/eye_mag/php/taskman_functions.php:73]:
```
text($task['to_name']) . " " . xla('via') . " " . text($task['to_fax']) . " " . xla('on') . " " . text($sent_date) ."'>
```

File: interface/forms/eye_mag/php/taskman_functions.php [hunk 4]
```diff
@@ -118,10 +127,10 @@ function process_tasks($task)
 
     if ($task['DOC_TYPE'] == "Fax") {
         //now return any objects you need to Eye Form
-        $send['DOC_link'] = "<a href='" . $webroot . "/openemr/controller.php?document&view&patient_id=" . attr($task['PATIENT_ID']) . "&doc_id=" . attr($task['DOC_ID']) . "'
-								target='_blank' title=" . xlt('Report was faxed. Click to view.') . ">
-								<i class='fa fa-file-pdf-o fa-fw'></i>
-							</a>";
+        $send['DOC_link'] = "<a onclick="openNewForm('" . $GLOBALS['webroot'] . "/controller.php?document&view&patient_id=" . attr($task['PATIENT_ID']) . "&doc_id=" . attr($task['DOC_ID']) . "', 'Fax Report');"
+                                href="JavaScript:void(0);"
+                                title='" . xlt('Report was faxed to') . " " . $task['to_name'] . " @ " . $task['to_fax'] . " on " .
```
Comment (bradymiller) [inline interface/forms/eye_mag/php/taskman_functions.php:132]:
```
title='" . xlt('Report was faxed to') . " " . text($task['to_name']) . " @ " . text($task['to_fax']) . " on " .
```

File: interface/forms/eye_mag/php/taskman_functions.php [hunk 5]
```diff
@@ -118,10 +127,10 @@ function process_tasks($task)
...
+                                $task['COMPLETED_DATE'] . ". " . xla(' Click to view.') . "'><i class='far fa-file-pdf fa-fw'></i></a>";
```
Comment (bradymiller) [inline interface/forms/eye_mag/php/taskman_functions.php:133]:
```
text($task['COMPLETED_DATE']) . ". " . xla(' Click to view.') . "'><i class='far fa-file-pdf fa-fw'></i></a>";
```

File: interface/forms/eye_mag/view.php [hunk 6]
```diff
@@ -4012,7 +4165,9 @@ class="PRIORS_class PRIORS"><i class="fa fa-spinner fa-spin"></i>
                                                   echo str_replace("form-control", "", $select_pharm);
                                                 ?>
                                               </td><td class="top">
-                                                  <button onclick="editScripts('/openemr/controller.php?prescription&list&id=<?php echo attr($pat_data['pid']); ?>');"><?php echo xlt('eRx'); ?></button>
+                                                  <a class="btn btn-primary btn-sm iframe rx_modal" href="javascript:;" onclick="editScripts('../../../controller.php?prescription&amp;list&amp;id=1')"><span>Edit</span></a>
```
Comment (bradymiller) [inline interface/forms/eye_mag/view.php:4168]: xlt() around Edit

File: sql/6_0_0-to-6_1_0_upgrade.sql [hunk 7]
```diff
@@ -191,6 +191,8 @@ INSERT INTO list_options (list_id,option_id,title, seq, is_default, option_value
 INSERT INTO list_options (list_id,option_id,title,seq,is_default,activity) VALUES ('Procedure_Billing','T','Third-Party',10,1,1);
 INSERT INTO list_options (list_id,option_id,title,seq,is_default,activity) VALUES ('Procedure_Billing','P','Self Pay',20,0,1);
 INSERT INTO list_options (list_id,option_id,title,seq,is_default,activity) VALUES ('Procedure_Billing','C','Bill Clinic',30,0,1);
+INSERT INTO list_options (list_id,option_id,title,seq,is_default,option_value,mapping,notes,codes,toggle_setting_1,toggle_setting_2,activity,subtype) VALUES ('Eye_Defaults_for_GENERAL', 'ODVITREOUS', 'clear', 504, 0, 0,'', 'RETINA','', 0, 0, 1,'');
+INSERT INTO list_options (list_id,option_id,title,seq,is_default,option_value,mapping,notes,codes,toggle_setting_1,toggle_setting_2,activity,subtype) VALUES ('Eye_Defaults_for_GENERAL', 'OSVITREOUS', 'clear', 506, 0, 0,'', 'RETINA','', 0, 0, 1,'');
```
Comment (bradymiller) [inline sql/6_0_0-to-6_1_0_upgrade.sql:195]: this needs to go to the bottom of this script in its own IF block

**Comment by bradymiller** [comment]
hi @ophthal , Just completed review on this. Very glad you are helping to fix this form up. Noted a couple minor issues above. Looking very forward to next update (goal is to get this fixed form in the next 6.0.0 patch).

**Comment by bradymiller** [comment]
@ophthal , Just let me know when this is ready. (I'll then do review/testing and likely get it into the codebase pretty quick since planning to get out 6.0.0 patch 1 after these fixes are in)

**Comment by ophthal** [comment]
I still have to do the reports section and taskman...

**Comment by bradymiller** [comment]
hi @ophthal , Review looks good (didn't test). You ok with me doing some work directly on this PR ? (couple minor fixes, fix merge conflict, and consolidate the third party packages into the eye form directory structure)

**Comment by ophthal** [comment]
Sure thing.
The more the merrier.
I understand the code is complex and anything to standardize it is fine.
I just don’t want to lose functionality and as you know it is packed...

**Comment by bradymiller** [comment]
cool. don't worry; not gonna change any of the code/functionality or third party libraries. just plan to migrate the third party libraries (and just the library files that are used) to a safe place so we never try to "upgrade" them again in the future :)

**Comment by bradymiller** [comment]
step 1: just brought in merge commit to fix conflict and update to most recent codebase

**Comment by bradymiller** [comment]
hi @ophthal , all done. Testing well for me (I just click on a bunch of stuff and nothing seems to break). My changes should have no impact on anything. I just did a couple minor things:
1. fixed psr12/css lint stuff
2. minor fix in sql upgrade
3. migrated the third party libraries to the eye form directory (this will prevent the community and me from trying to upgrade/update/remove these libraries :) )
4. couple minor html escaping things

Gonna bring this in after some more testing (definitely let me know if i messed up something) and then will get it into rel-600 branch and the  patch 1 for 6.0.0 . Thanks for doing all this and thanks for fixing all the things we broke! -brady

**Pull request merged**